ports/opus/portfile.cmake

```
@@ -17,16 +18,18 @@ if(VCPKG_TARGET_IS_MINGW)
     set(STACK_PROTECTOR OFF)
     string(APPEND VCPKG_C_FLAGS "-D_FORTIFY_SOURCE=0")
     string(APPEND VCPKG_CXX_FLAGS "-D_FORTIFY_SOURCE=0")
-    if(VCPKG_TARGET_ARCHITECTURE MATCHES "^(ARM|arm)64$")
-        list(APPEND ADDITIONAL_OPUS_OPTIONS "-DOPUS_USE_NEON=OFF") # for version 1.3.1 (remove for future Opus release)
-        list(APPEND ADDITIONAL_OPUS_OPTIONS "-DOPUS_DISABLE_INTRINSICS=ON") # for HEAD (and future Opus release)
-    endif()
 elseif(VCPKG_TARGET_IS_EMSCRIPTEN)
     set(STACK_PROTECTOR OFF)
 else()
     set(STACK_PROTECTOR ON)
 endif()
 
+# Fix build on mingw arm64-mingw-*, arm-linux and arm-android
+if(VCPKG_TARGET_TRIPLET MATCHES "^(arm64-mingw|arm-linux|arm-android)")
```

Don't match target triplet names. They are just for human readers, and user might choose different custom names.

Not perfect, but good enough IMO:
```suggestion
if((VCPKG_TARGET_IS_MINGW AND VCPKG_TARGET_ARCHITECTURE MATCHES "^arm")
   (VCPKG_TARGET_IS_LINUX AND VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
   (VCPKG_TARGET_IS_ANDROID AND VCPKG_TARGET_ARCHITECTURE STREQUAL "arm" AND VCPKG_CMAKE_CONFIGURE_OPTIONS MATCHES "ANDROID_ARM_NEON"))
```
(If we don't assume NEON support for arm64-mingw, we probably also don't want it for arm-mingw, if it exists at all.)

The alternative is to move initialization to a `CMAKE_PROJECT_INCLUDE`. This would run in project mode right after toolchain initialization, so it wouldn't need to match `VCPKG_CMAKE_CONFIGURE_OPTIONS` for detecting android arm neon.

---

I like this alternative because the if-clause is bulky

---

> I like this alternative because the if-clause is bulky

We simply must not match target triplet names.
It only gets simpler if upstream gets smarter.